rosa/cli.py

```diff
@@ -157,99 +150,54 @@ def get_available_flags(command):
 
 @functools.cache
 def parse_help(rosa_cmd="rosa"):
-    commands_dict = {}
+    commands_dict = benedict.benedict()
+
+    def _fill_commands_dict_with_support_flags(flag_key_path):
+        support_commands = {
+            "json_output": "-o, --output",
+            "auto_answer_yes": "-y, --yes",
+            "auto_mode": "-m, --mode",
+            "region": "--region",
+        }
+        for cli_flag, flag_value in support_commands.items():
+            commands_dict[flag_key_path][cli_flag] = check_flag_in_flags(
+                command_list=["rosa"] + flag_key_path,
+                flag_str=flag_value,
+            )
+
     _commands = get_available_commands(command=[rosa_cmd])
-    output_flag_str = "-o, --output"
-    auto_answer_yes_str = "-y, --yes"
-    auto_mode_str = "-m, --mode"
-    region_str = "--region"
 
     for command in _commands:
         commands_dict.setdefault(command, {})
 
     for top_command in commands_dict.keys():
-        _commands = get_available_commands(command=[rosa_cmd, top_command])
+        flag_search_path = [top_command]
+        _commands = get_available_commands(command=["rosa"] + flag_search_path)
 
         if _commands:
             # If top command has sub command
             for command in _commands:
-                commands_dict[top_command][command] = {}
-                _commands = get_available_commands(
-                    command=[rosa_cmd, top_command, command]
-                )
+                flag_search_path = [top_command, command]
```

Comment:
```suggestion
                flag_search_path.append(command)
```

Reply:
Append affect the functionality since it make `flag_search_path` to hold the last command from the last loop iteration.

Example:
Desired output:
'create': {
'account-roles': {'json_output': False, 'auto_answer_yes': True, 'auto_mode': True, 'region': False}, 
'admin': {'json_output': True, 'auto_answer_yes': True, 'auto_mode': False, 'region': True}
...}

`flag_search_path` was ['create','account-roles'] and THEN ['create','admin']

Output With append:
`flag_search_path` = ['create','account-roles','admin']

'create': {
'account-roles': {
'json_output': False, 'auto_answer_yes': True, 'auto_mode': True, 'region': `False, 'admin':` {'json_output': False, 'auto_answer_yes': True, 'auto_mode': True, 'region': `False, 'cluster': `{
..}}}}}}

Reply:
@rnetser

Reply:
so please rename arg, do not overwrite the top one with something else